test/test_upgrade_step_check_init_inst.py

```diff
@@ -0,0 +1,116 @@
+import unittest
+from hamcrest import assert_that, equal_to, is_not
+from mock import MagicMock, patch, mock_open
+from mother import LoggingStub, FileAccessStub
+from src.upgrade_step_check_init_inst import UpgradeStepCheckInitInst
+import os
+
+import sys
+module_ = "builtins"
+module_ = module_ if module_ in sys.modules else '__builtin__'
+
+try:
+    import unittest.mock as mock
+except (ImportError,) as e:
+    import mock 
+
+class TestUpgradeStepCheckInitInst(unittest.TestCase):
+
+    def setUp(self):
+        self.file_access = FileAccessStub()
+        self.upgrade_step = UpgradeStepCheckInitInst()
+        self.logger = LoggingStub()
+        self.directory_structure = [
+            ("root", ["dir1", "dir2"], ["init_file1"]),
+            ("root\\dir1", ["dir3"], ["init_file2", "init_file3"]),
+            ("root\\dir2", [], ["init_file4"]),
+            ("root\\dir3", [], [])
+        ]
+
+    def test_GIVEN_no_file_called_init_inst_WHEN_search_files_THEN_zero_returned(self):
+        # Arrange
+        file_names = ["init", "my_init", "another_file"]
+        root = "myfolder"
+        # Act and Assert
+        assert_that(self.upgrade_step.search_files(file_names, root, self.file_access),
+            equal_to(0), "File starting with init_inst is not in filenames, so no pre or post cmd would be found by genie")
+    
+    def test_GIVEN_no_file_called_init_inst_containing_precmd_WHEN_search_files_THEN_zero_returned(self):
```

Comment:
This is exactly the same as the test above

Reply:
Yes, not sure what was meant by the test. Removed.